api/lcp/collection.py

```diff
@@ -0,0 +1,311 @@
+
+        :param pin: The patron's alleged password
+        :type pin: string
+
+        :param licensepool: Contains lending info as well as link to parent Identifier
+        :type licensepool: LicensePool
+
+        :param internal_format: A vendor-specific name indicating the format requested by the patron
+        :type internal_format:
+
+        :param part: A vendor-specific identifier indicating that the
+            patron wants to fulfill one specific part of the book
+            (e.g. one chapter of an audiobook), not the whole thing
+        :type part: Any
+
+        :param fulfill_part_url: A function that takes one argument (a
+            vendor-specific part identifier) and returns the URL to use
+            when fulfilling that part
+        :type fulfill_part_url: Any
+
+        :return: a FulfillmentInfo object
+        :rtype: FulfillmentInfo
+        """
+        loan = get_one(self._db, Loan, patron=patron, license_pool=licensepool, on_multiple='interchangeable')
+        license = self._lcp_server.get_license(loan.external_identifier, loan.external_identifier)
+        fulfillment_info = LCPFulfilmentInfo(
+            licensepool.identifier.identifier,
+            licensepool.collection,
+            licensepool.data_source.name,
+            licensepool.identifier.type,
+            content_link=None,
+            content_type=DeliveryMechanism.LCP_DRM,
+            content=license,
+            content_expires=None,
+        )
+
+        return fulfillment_info
+
+    def patron_activity(self, patron, pin):
+        """Returns patron's loans
+
+        :param patron: A Patron object for the patron who wants to check out the book
+        :type patron: Patron
+
+        :param pin: The patron's alleged password
+        :type pin: string
+
+        :return: List of patron's loans
+        :rtype: List[LoanInfo]
+        """
+        # NOTE: We assume that there can be only ONE LCP collection per library
+        loans = self._db\
+            .query(Loan)\
+            .join(LicensePool)\
+            .join(Collection)\
+            .filter(
+                Loan.patron == patron,
+                Collection.id == self._collection_id
+            )
+
+        loan_info_objects = []
+
+        for loan in loans:
+            licensepool = get_one(self._db, LicensePool, id=loan.license_pool_id)
+
+            loan_info_objects.append(LoanInfo(
+                collection=self.collection,
+                data_source_name=licensepool.data_source.name,
+                identifier_type=licensepool.identifier.type,
+                identifier=licensepool.identifier.identifier,
+                start_date=loan.start,
+                end_date=loan.end,
+                fulfillment_info=None,
+                external_identifier=loan.external_identifier
+            ))
+
+        return loan_info_objects
+
+    def place_hold(self, patron, pin, licensepool, notification_email_address):
+        pass
```

Comment:
Add a comment explaining whether we expect to implement this later.

Reply:
Could you tell me what would happen if I don't implement those methods? Circulation Manager will be throwing exceptions when there are no available licenses?

Reply:
Yes, I think you'll get unhandled exceptions, because `CirculationAPI` assumes these methods either raise an exception or return a `HoldInfo`. (This is a problem in enki.py as well, now that I look at it... I'm not sure if Enki books ever lose all their available licenses though.)

If you plan to add this later then I think let these methods inherit the `NotImplementedError` from `BaseCirculationAPI` and file a ticket to finish the work.

Reply:
Removed the methods and added a comment